File: task-03/Haskell/subtask-03.hs
import Control.Monad (forM_)
import Text.Printf (printf)

printStarPattern :: Int -> IO ()
printStarPattern n = do
    forM_ [1,3..n] $ \i -> do
        let padding = replicate ((n - i) `div` 2) ' '
        let stars = replicate i '*'
        putStrLn (padding ++ stars)

    forM_ [n-2,n-4..1] $ \i -> do
        let padding = replicate ((n - i) `div` 2) ' '
        let stars = replicate i '*'
        putStrLn (padding ++ stars)

main :: IO ()
main = do
    putStr "Enter a number: "
    input <- getLine
    let n = read input :: Int
    printStarPattern(n)

File: task-03/C/subtask-03.c
#include <stdio.h>

void print_star_pattern(int n) {
    for (int i = 1; i <= n; i += 2) {
        for (int j = 0; j < (n - i) / 2; j++) {
            printf(" ");
        }
        for (int j = 0; j < i; j++) {
            printf("*");
        }
        printf("\n");
    }

    for (int i = n - 2; i >= 1; i -= 2) {
        for (int j = 0; j < (n - i) / 2; j++) {
            printf(" ");
        }
        for (int j = 0; j < i; j++) {
            printf("*");
        }
        printf("\n");
    }
}

int main() {
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);

    print_star_pattern(n);

    return 0;
}
